assets/src/components/selection/datetime-select/index.js

```diff
@@ -0,0 +1,231 @@
+/**
+ * External dependencies
+ */
+import { stringify } from 'querystringify';
+import moment from 'moment';
+import { isUndefined, pickBy, isEmpty } from 'lodash';
+import PropTypes from 'prop-types';
+import { __ } from '@eventespresso/i18n';
+
+/**
+ * WordPress dependencies
+ */
+import { Placeholder, SelectControl, Spinner } from '@wordpress/components';
+import { withSelect } from '@wordpress/data';
+import { Component, Fragment } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { datetimeSelectOptions } from './options';
+
+const nowDateAndTime = moment();
+
+/**
+ * DatetimeSelect component.
+ * A react component for an datetime selector.
+ *
+ * @param {Array} datetimes  An empty array or array of Datetime Entities. See
+ *                          prop-types for shape.
+ * @param {function} onDatetimeSelect  The callback on selection of datetime.
+ * @param {string} selectLabel The label for the select input.
+ * @param {number} selectedDatetimeId  If provided, the id of the datetime to
+ *   pre-select.
+ * @param {number} forEventId   ID for Event to retrieve datetimes from
+ * @param {boolean} isLoading  Whether or not the selector should start in a
+ *                               loading state
+ * @return {Function}  A pure component function.
+ * @constructor
+ */
+class DatetimeSelect extends Component
+{
+	constructor() {
+		super( ...arguments );
+		this.placeHolder = this.placeHolder.bind( this );
+	}
+
+	placeHolder() {
+		const { isLoading, selectLabel } = this.props;
+		return (
+			<Fragment>
+				<Placeholder
+					icon="calendar"
+					label={ selectLabel }
+				>
+					{ ! isLoading ?
+						__(
+							'There are no datetimes to select from. You need' +
+							' to create a datetime first.',
+							'event_espresso',
+						) :
+						<Spinner/>
```

Comment:
I _think_ there may be an .eslint error due to there being no space before the closing bracket.  In other words, this needs to be: `<Spinner />`